### DISCUSSION TITLE
Request: Streaming completions

### DISCUSSION THREAD

**Discussion by 3inary** [srcnalt/OpenAI-Unity#22] [Ideas]
Thanks for this well crafted package!

I have a request, could you please add Streaming to completions?
As described in the API:
stream boolean Optional Defaults to false
Whether to stream back partial progress. If set, tokens will be sent as data-only [server-sent events] as they become available, with the stream terminated by a data: [DONE] message.

With media type "application/octet-stream"

**Comment by srcnalt** [comment]
Hi, @3inary I made this work on a test, but this will require some refactoring of the current request dispatcher. I can bring this over this week.

**Comment by Morgan-6Freedom** [comment]
Hello. This would be awesome because that's kind of complex to do in Unity with UnityWebRequest. Did you succeed to do it @srcnalt ?

**Comment by srcnalt** [comment]
Hi @Morgan-6Freedom, @3inary 

Just released streaming support for text and chat completion endpoints. Tbh, I'm not super happy with how I had to implement it but works as intended. You can find example usage in the readme and in endpoint tests. Also a sample added for this feature.
